### ISSUE TITLE
Confusion/bug on volume rendering with mayavi

### ISSUE THREAD

**Issue by 22csnyder** [enthought/mayavi#244]
Hi,

I wanted to bring attention to a bug I've documented on stackoverflow:
http://stackoverflow.com/questions/32102213/mayavi-enthought-volume-rendering-buggy-gui

Sorry if you were already aware of this. It is literally bringing my project to a standstill because users can't adjust the opacity or ctf.

Chris

**Comment by 22csnyder** [comment]
I have posted the following reasonable workaround.
http://stackoverflow.com/questions/32102213/mayavi-enthought-volume-rendering-buggy-gui/32194439#32194439

The problem is vexing though. Let me know if anyone tries to tackle this issue and I'll do what I can to help. I found following the chain of events that happen when you change the default ctf editor to be hard to follow.

My best guess is that there are some default settings, like red-blue fade, that normally do not come into play unless certain traits are changed, overwriting the normal blue-green-red.

Some strange things happen with the data range. When switching data scalled [0.03,1] with data [0.05,2] (approximations), sometimes the ctf range would change wildly to [0,40]. Sometimes all the points are thrown to the right of the ctf on the gui, as shown on the video. This is probably the opposite phenomenon.

Best,
Chris S

**Comment by jgodwin** [comment]
I ran into a very similar issue. I was able to work around it by rescaling all of my data to the range [0,255] exactly AND converting it from a 32-bit float datatype to a 32-bit integer datatype. Once you do that, the CTF seems to hold it's values as expected after setting the data.
